tests/integration_tests/data_sources/test_maria_ds.py

```diff
@@ -0,0 +1,62 @@
+import mysql.connector
+import datetime
+import logging
+from mindsdb import Predictor
+from mindsdb.libs.constants.mindsdb import DATA_TYPES, DATA_SUBTYPES
+from mindsdb.libs.data_sources.maria_ds import MariaDS
+
+
+def test_maria_ds():
+    HOST = 'localhost'
+    USER = 'root'
+    PASSWORD = ''
+    DATABASE = 'mysql'
+    PORT = 4306
+
+    con = mysql.connector.connect(host=HOST,
+                                  port=PORT,
+                                  user=USER,
+                                  password=PASSWORD,
+                                  database=DATABASE)
+    cur = con.cursor()
+
+    cur.execute('DROP TABLE IF EXISTS test_mindsdb')
+    cur.execute('CREATE TABLE test_mindsdb(col_categorical Text, col_int BIGINT, col_bool BOOL, col_float FLOAT, col_date Text, col_ts Text, col_text Text)')
```

Comment:
Please check for the types: Date, Datetime, Timestamp, Time (see here: https://mariadb.com/kb/en/data-types/)

I think python datetime object will fit in the last 3 and the former should fit a python date (insert-able via calling `.date()` on  a datetime object) But you'd need to insert via the template syntax rather than using string format.

If template syntax doesn't work just test whichever are easy to test inserted this way.